processor/metricstransformprocessor/factory.go

```diff
@@ -96,6 +97,43 @@ func validateConfiguration(config *Config) error {
 			}
 		}
 	}
-
 	return nil
 }
+
+// fillMappingData constructs the map that will be useful for the operations and stores them in the config
+func fillMappingData(config *Config) {
+	for i := range config.Transforms {
+		t := &config.Transforms[i]
+		for j := range t.Operations {
+			op := &t.Operations[j]
+			if len(op.ValueActions) > 0 {
+				op.ValueActionsMapping = createLabelValueMapping(op.ValueActions)
+			}
+			if op.Action == AggregateLabels {
+				op.LabelSetMap = sliceToSet(op.LabelSet)
+			} else if op.Action == AggregateLabelValues {
+				op.LabelSetMap = map[string]bool{op.Label: true}
+				op.AggregatedValuesSet = sliceToSet(op.AggregatedValues)
+			}
+		}
+	}
+}
+
+// createLabelValueMapping creates the labelValue rename mappings based on the valueActions
+func createLabelValueMapping(valueActions []ValueAction) map[string]string {
```

Comment:
Does the order of the value actions specified in the configuration matter? If not, why not change the configuration to be a map that way it gets parsed into a map structure?

Reply:
Thanks for the feedback, I think the current configuration with `Value` and `NewValue` fields are explicit to users about what to fill out here in the config. But this is a great idea and I am open to having this changed as well.

Reply:
👍  Would need to consider this in light of the rest of the config entrires as well as you want it to look consistent.

Side note: I think you are missing an example for `value_actions` in the README

Reply:
It isn't very clear to me why value_actions would need to have more than one entry so an example would be really helpful. I view it as if there are multiple values that need ot be updated, there should be multiple operations.